pkg/datasource/schedule_with_details.go

```diff
@@ -136,11 +136,21 @@ func (schedule *Schedule) UpdateNextReportTime() {
 			reportTime = reportTime.AddDate(0, 1, daysOffset)
 		}
 	case 2: // fortnightly
-		daysToAdd := (scheduleDays - int(reportTime.Day()) + 14) % 14
-		reportTime = reportTime.AddDate(0, 0, daysToAdd)
+		if scheduleDays == int(reportTime.Weekday()) {
```

Comment:
@anildahalsussol there is a bug here. You are checking Weekday for fortnightly. :D
Too bad I saw this late.